tests/test_transforms/test_base/test_base.py

```diff
@@ -0,0 +1,95 @@
+from unittest.mock import Mock
+
+import pandas as pd
+import pytest
+
+from etna.datasets import TSDataset
+from etna.datasets import generate_ar_df
+from etna.transforms import NewTransform
+
+
+class NewTransformMock(NewTransform):
+    def _fit(self, df: pd.DataFrame) -> "NewTransform":
+        return self
+
+    def _transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        return df
+
+
+@pytest.fixture
+def remove_columns_df():
+    df = generate_ar_df(periods=10, n_segments=3, start_time="2000-01-01")
+    df["exog_1"] = 1
+    df = TSDataset.to_dataset(df)
+
+    df_transformed = generate_ar_df(periods=10, n_segments=3, start_time="2000-01-01")
+    df_transformed = TSDataset.to_dataset(df_transformed)
+    return df, df_transformed
+
+
+@pytest.mark.parametrize(
+    "in_column, expected_features",
+    [("all", "all"), ("target", ["target"]), (["target", "segment"], ["target", "segment"])],
+)
+def test_required_features(in_column, expected_features):
+    transform = NewTransformMock(in_column=in_column)
+    assert transform.required_features == expected_features
+
+
+def test_update_dataset_remove_columns(remove_columns_df, expected_features_to_remove=["exog_1"]):
```

Comment:
It is strange to see info about removed features in the second parameter because it is related to the first parameter.

Reply:
I don't get the problem here)

Reply:
Move computation of this parameter inside the test